test/plugins/tokens.test.js

```diff
@@ -218,24 +225,119 @@ describe('token plugin test', () => {
+
+        it('returns 404 when the user does not exist', () => {
+            userFactoryMock.get.withArgs({ username }).resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 409 when a token with the same name already exists', () => {
+            options.url = `/tokens/${tokenId + 1}`;
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 409);
+                assert.strictEqual(reply.result.message,
+                    `Token with name ${testToken.name} already exists`);
+            });
+        });
+
+        it('returns 200 if update successfully', () => {
+            const expected = hoek.applyToDefaults(testToken, {
+                description: 'a new description'
+            });
+
+            tokenMock.toJson.returns(expected);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 200);
+                assert.calledOnce(tokenMock.update);
+                assert.deepEqual(reply.result, expected);
+            });
+        });
+
+        it('doesn\'t error if the new name conflicts only with the old name', () => {
+            const expected = hoek.applyToDefaults(testToken, {
+                description: 'a new description'
+            });
+
+            tokenMock.toJson.returns(expected);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 200);
+                assert.calledOnce(tokenMock.update);
+                assert.deepEqual(reply.result, expected);
+            });
+        });
+
+        it('returns 403 when the user does not own the token', () => {
             userFactoryMock.get.withArgs({ username }).resolves({
-                id: testToken.userId
+                id: testToken.userId + 1
             });
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 403);
+            });
+        });
+
+        it('returns 500 when the token model fails to update', () => {
+            const testError = new Error('tokenModelUpdateError');
+
+            tokenMock.update.rejects(testError);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 500);
+            });
+        });
+    });
+
+    describe('DELETE /tokens/{id}', () => {
+        let options;
+
+        beforeEach(() => {
+            options = {
+                method: 'DELETE',
+                url: `/tokens/${tokenId}`,
+                credentials: {
+                    username,
+                    scope: ['user']
+                }
+            };
+
+            tokenFactoryMock.get.resolves(tokenMock);
+            userMock.tokens = tokensGetterMock([tokenMock]);
+            userFactoryMock.get.withArgs({ username }).resolves(userMock);
```

Comment:
Ah, there's where my confusion was. I thought that beforeEach ran before each of the `describe` blocks. Didn't realize it also ran before `it` blocks, so I thought I needed to have that there to restore functionality after doing things like having the `userFactoryMock` resolve `null` to test the 404 case.